ext/node/resolution.rs

```diff
@@ -705,12 +707,13 @@ pub fn legacy_main_resolve(
   package_json: &PackageJson,
   referrer_kind: NodeModuleKind,
 ) -> Result<PathBuf, AnyError> {
-  let maybe_main =
-    if referrer_kind == NodeModuleKind::Esm && package_json.typ == "module" {
-      &package_json.module
```

Comment:
Huge nitpick and feel free to ignore: I think it's easier to understand if this were kept mostly the same, but then `package_json.module.as_ref().or_else(|| package_json.module.as_ref())` to avoid the `let mut`.